contracts/modules/STO/USDTiered/USDTieredSTO.sol

```diff
@@ -698,6 +698,7 @@ contract USDTieredSTO is USDTieredSTOStorage, STO {
         return (tiers[tiers.length - 1].totalTokensSoldInTier == tiers[tiers.length - 1].tokenTotal);
     }
 
+    // CONCERN: This function should be removed or change there access modifier to internal in the favor of MakerDao Oracles V2
```

Comment:
Is there something clever we can do where the function will always revert if called on-chain, but can still be called off-chain as a view function? This would mean that no-one can use the USDTieredSTO as a way to bypass Maker's whitelist, whilst still allowing the UI etc. to access this rate.

Reply:
One way will be to read the makerdao oracle's value offchain and then send it as a parameter to this function.
Onchain, we'll need a separate internal function that fetches the value from the oracle.

btw, Why do we even want to limit this function from being called onchain? Does makerdao now charge a fee based on the number of calls made? In their medianizer contract, it's a  simple whitelist check but I am not sure what the offchain licensing terms are.

Reply:
I think the clever way is to use msg.sender as 0x0